sql/moz-fx-data-shared-prod/telemetry_derived/adm_engagements_daily_v1/query.sql

```diff
@@ -0,0 +1,198 @@
+      'bug-1678671-pref-topsites-url-redirect-germany-launch-v2-release-80-82') IS NOT NULL THEN mozdata.udf.get_key(event_map_values,
+      'source')
+      WHEN mozdata.udf.get_key(experiments, 'bug-1657447-pref-topsites-url-redirect-uk-launch-v5-release-80-80') IS NOT NULL THEN mozdata.udf.get_key(event_map_values, 'source')
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678673-pref-topsites-url-redirect-uk-launch-v5-v2-release-80-82') IS NOT NULL THEN mozdata.udf.get_key(event_map_values,
+      'source')
+    ELSE
+    event_object
+  END
+    AS placement,
+    CASE
+      WHEN mozdata.udf.get_key(experiments, 'bug-1655185-pref-topsites-redirect-launch-us-v5-release-80-80') IS NOT NULL THEN mozdata.udf.get_key(event_map_values, 'partner')
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678672-pref-topsites-redirect-launch-us-v5-v2-release-80-82') IS NOT NULL THEN mozdata.udf.get_key(event_map_values,
+      'partner')
+      WHEN mozdata.udf.get_key(experiments, 'bug-1657450-pref-topsites-url-redirect-germany-launch-release-80-80') IS NOT NULL THEN mozdata.udf.get_key(event_map_values, 'partner')
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678671-pref-topsites-url-redirect-germany-launch-v2-release-80-82') IS NOT NULL THEN mozdata.udf.get_key(event_map_values,
+      'partner')
+      WHEN mozdata.udf.get_key(experiments, 'bug-1657447-pref-topsites-url-redirect-uk-launch-v5-release-80-80') IS NOT NULL THEN mozdata.udf.get_key(event_map_values, 'partner')
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678673-pref-topsites-url-redirect-uk-launch-v5-v2-release-80-82') IS NOT NULL THEN mozdata.udf.get_key(event_map_values,
+      'partner')
+    ELSE
+    event_string_value
+  END
+    AS partner,
+    CASE
+      WHEN mozdata.udf.get_key(experiments, 'bug-1655185-pref-topsites-redirect-launch-us-v5-release-80-80') IS NOT NULL THEN event_object
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678672-pref-topsites-redirect-launch-us-v5-v2-release-80-82') IS NOT NULL THEN event_object
+      WHEN mozdata.udf.get_key(experiments, 'bug-1657450-pref-topsites-url-redirect-germany-launch-release-80-80') IS NOT NULL THEN event_object
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678671-pref-topsites-url-redirect-germany-launch-v2-release-80-82') IS NOT NULL THEN event_object
+      WHEN mozdata.udf.get_key(experiments, 'bug-1657447-pref-topsites-url-redirect-uk-launch-v5-release-80-80') IS NOT NULL THEN event_object
+      WHEN mozdata.udf.get_key(experiments,
+      'bug-1678673-pref-topsites-url-redirect-uk-launch-v5-v2-release-80-82') IS NOT NULL THEN event_object
+    ELSE
+    event_method
+  END
+    AS interaction,
+    SPLIT(app_version, '.')[
+  OFFSET
+    (0)] AS version,
+    normalized_channel,
+    CASE
+    -- phase 2
+      WHEN mozdata.udf.get_key(experiments, 'bug-1655185-pref-topsites-redirect-launch-us-v5-release-80-80') IS NOT NULL THEN 'phase2'
```

Comment:
I'd like to avoid the repetition of these experiment slugs. I think we can simplify by creating the `phase` column in a CTE, then filtering to only rows with a non-null `phase`. AFAICT, this would prevent all of the other enumerations of these experiment slugs.

Here's a quick attempt at a UDF that might reduce some of the repetition in creating the `phase` column itself.

```sql
CREATE OR REPLACE FUNCTION map.contains_any_of(map ANY TYPE, k ANY TYPE) AS (
  SELECT
    LOGICAL_OR(key_value IN k)
  FROM
    UNNEST(map) AS key_value
);
```

Reply:
oh cool! 😎 
can you use the function on the temp table for the case statements, there's the event schema change that puts a wrinkle in it. would be cool to have it phase dependent instead of the repeated slugs as you mentioned